frontend/src/scenes/trends/Trends.tsx

```diff
@@ -22,6 +23,8 @@ export function TrendInsight({ view }: Props): JSX.Element {
     )
     const { loadMoreBreakdownValues } = useActions(trendsDataLogic(insightProps))
 
+    const { dataNodeProps } = useValues(dataNodeLogic)
```

Comment:
A problem I was noticing when I used the `TrendsQuery` insight viz was that the `trendsFilter: { display }` option didn't seem to be passing in the `display` value correctly to this component. The `display` value from `trendsDataLogic` would consistently return undefined even when I passed in a value, however, `dataNodeProps` held the display value correctly so I wonder if something is misaligned here with the key value logic match when we dive into where the `display` value comes from in `trendsDataLogic`

Reply:
Oh hmm, sorry that this isn't working yet :( Did some digging and the following is happening here:

Currently the `<Query />` node for `InsightVizQueries` was built as replacement for the existing functionality. The code is still tightly coupled with the internals of insights and seems to not work for standalone queries.

More concretely the query gets overwritten by the default trends query that insights fall back to. This doesn't have a `display` set and thus you get `undefined`.

I think I have a fix ready here https://github.com/PostHog/posthog/pull/16741, but need to do more testing tomorrow.

Reply:
cool! i'm happy to test and review too, feel free to ping me on anything 😃